### DISCUSSION TITLE
How do I change (wrong) default screenshot size

### DISCUSSION THREAD

**Discussion by Seyen** [ImageMagick/ImageMagick#5061] [Help] [answered]
Running `identify screenshot:` on Win10 returns `screenshot:=> SCREENSHOT 1536x864 1536x864+0+0 16-bit sRGB 0.016u 0:00.045`, but my screen resolution is `1920x1080` (and I don't use a second monitor). This of course means that any screenshots taken this way returns a "cropped" image of whatever is on my screen. 

I've tried a lot of options to get the full screen(size, geometry, page, repage, others, in increasingly ridiculous fashion), but all of them only manipulate the size of the original 1536x864 image.

Could someone please advise how to adjust whatever config governs this?

**Comment by snibgo** [comment]
It works for me, with IM v7.1.0-4, pre-built, on Windows 8.1, with a 1920x1080 screen:
```
f:\web\im>%IMG7%magick screenshot: info:
screenshot: SCREENSHOT 1920x1080 1920x1080+0+0 16-bit sRGB 0.000u 0:00.002
```

I can specify a rectangle to grab like this:
```
magick screenshot:[100x100+230+340] x.png
```
... but I can't grab a rectangle that exceeds my screen size.

The code in coders\screenshot.c looks reasonable:
```
      geometry.width=(size_t) GetDeviceCaps(hDC,HORZRES);
      geometry.height=(size_t) GetDeviceCaps(hDC,VERTRES);
```
IM asks Windows how large the screen is.

**Comment by Seyen** [comment, marked as answer]
EDIT: I've found a workaround that will work for now: open up properties for magick.exe and set compatibility > Change high DPI settings > High DPI Scaling Override > Application, this gets the proper size for the screenshot format.

@snibgo I'm assuming that the implication in your answer is "There is no screen resolution setting, it's supposed to be the right size". Just had a thought. I'm using the (Windows native) DPI scaling option on my machine, and the scaling is 125%, and well, turns out 1536x864 scaled by 1.25 is 1920x1080.

Is there perhaps some DPI scaling option in ImageMagick I can use to influence the screenshot pseudo format?

**Comment by snibgo** [comment]
It would seem reasonable that Windows can tell applications that the screen has fewer pixels than it really has, to make the application draw its objects larger. And this should be overridable, either at the OS level, or within applications. It seems to be overridable at the OS level by "Disable display scaling on high DPI settings" for any executables, but is not (as far as I know) overridable within applications.

I did some tests on my laptop, under Windows 8.1.

In Control Panel, Display, there is an option to "Change the size of all items". They can be "smaller" or "larger". Mine is set to the smallest, which is also the "Recommended size", which I think means my screen has that many physical pixels. When I set this to "larger", everything is larger but blurrier. IM "screenshot:" makes an image that is 1280x720, and doesn't grab all the pixels from my screen. The resulting image is truncated at the right and bottom.

There is a similar option to "Change the appearance of your display", with resolution settings from 1920x1080 to 1024x768. When I set this to 1280x720, everything is larger but blurrier. IM "screenshot:" makes an image that is 1280x720 as before, but this time it grabs the entire screen, and doesn't truncate the image.

I don't know what Windows is doing to make one of these truncate, and the other not.

When I repeat the experiment with "Disable display scaling on high DPI settings" for magick.exe, IM "screenshot:" now makes an image that is 1920x1080, and captures the entire screen. So I think that is the correct solution for you. If you use a graphic editor such as Gimp, you might want to consider doing the same thing for gimp.exe, to aid evaluation of blurriness/sharpness of images.

See also https://answers.microsoft.com/en-us/windows/forum/all/how-to-disable-display-scaling-on-high-dpi-setting/c546cd4c-c4b9-4cb0-8e7b-52fb8906ef70 and many similar queries on the net.
